src/nvim/api/private/helpers.c

```diff
@@ -1377,6 +1377,9 @@ Object copy_object(Object obj)
     case kObjectTypeArray:
       return ARRAY_OBJ(copy_array(obj.data.array));
 
+    case kObjectTypeBuffer:
+    case kObjectTypeTabpage:
+    case kObjectTypeWindow:
```

Comment:
these EXT types are aliases to `handle_T` which is just `int`. So I would think these cases should be handled like `kObjectTypeInteger` (line 1371), not `kObjectTypeDictionary`.

Reply:
That also explains the weird bug I was seeing in Firenvim 🤦.